File: packages/create-react-native-library/templates/java-library-mixed/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}Module.java
package com.<%- project.package %>;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactMethod;

public class <%- project.name -%>Module extends <%- project.name -%>Spec {
  public static final String NAME = "<%- project.name -%>";

  <%- project.name -%>Module(ReactApplicationContext context) {
    super(context);
  }

  @Override
  @NonNull
  public String getName() {
    return NAME;
  }

<% if (project.cpp) { -%>
  static {
    System.loadLibrary("cpp");
  }

  public static native double nativeMultiply(double a, double b);
<% } -%>

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  @ReactMethod
  public void multiply(double a, double b, Promise promise) {
<% if (project.cpp) { -%>
    promise.resolve(nativeMultiply(a, b));
<% } else { -%>
    promise.resolve(a * b);
<% } -%>
  }
}

File: packages/create-react-native-library/templates/kotlin-view-library-mixed/android/src/main/java/com/{%- project.package_dir %}/{%- project.name %}Module.kt
package com.<%- project.package %>

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class <%- project.name -%>Module internal constructor(context: ReactApplicationContext) :
  <%- project.name -%>Spec(context) {

  override fun getName(): String {
    return NAME
  }

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  @ReactMethod
  override fun multiply(a: Double, b: Double, promise: Promise) {
    promise.resolve(a * b)
  }

  companion object {
    const val NAME = "<%- project.name -%>"
  }
}
